BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeClient.java

```diff
@@ -76,7 +78,11 @@ private static BraintreeClientParams createDefaultParams(Context context, String
      * @param authorization The tokenization key or client token to use. If an invalid authorization is provided, a {@link BraintreeException} will be returned via callback.
      */
     public BraintreeClient(@NonNull Context context, @NonNull String authorization) {
-        this(createDefaultParams(context, authorization));
+        this(createDefaultParams(context, authorization, null));
+    }
+
+    public BraintreeClient(@NonNull Context context, @NonNull ClientTokenProvider clientTokenProvider) {
```

Comment:
I think it's fine to handle deprecation of old methods in another PR, but let's add doc strings for these new ones now so we don't miss any that are added

Reply:
Done!